src/components/SurveyComponent.vue

```diff
@@ -1,324 +1,277 @@
+        client.request('/QuestionnaireResponse?patient=' +  this.patientId)
+      ];
+      //get all resources
+      Promise.all(requests).then(results => {
+        results.forEach(result => {
+          if (!result) return true;
+          if (result.resourceType == 'Bundle' && result.entry) {
+            result.entry.forEach(o => {
+              if (o && o.resource) this.patientBundle.entry.push({resource: o.resource});
+            });
+          } else if (Array.isArray(result)) {
+            result.forEach(o => {
+              if (o.resourceType) this.patientBundle.entry.push({resource: o});
+            });
+          } else {
+            this.patientBundle.entry.push({resource: result});
+          }
+        });
+      });
+    }, //
+    getQuestionnaireURL() {
+      if (!this.questionnaire) return null;
+      //used to pair questionaire with responses
+      return this.questionnaire.url;
+    },
+    setQuestionnaireAuthor() {
+      // Add the `subject` element to the QuestionnaireResponse
+      this.questionnaireResponse.subject = {
+        reference: `Patient/${this.patientId}`
+      };
 
-    // Get any Procedure resources
-    await client.request('/Procedure?patient=' + pid).then(function(pr) {
-      if (pr) {
-        if (pr.resourceType == 'Bundle' && pr.entry) {
-          pr.entry.forEach(p => {
-            if (p.resource) patientBundle.entry.push({resource: p.resource});
-          });
-        } else if (Array.isArray(pr)) {
-          pr.forEach(p => {
-            if (p.resourceType) patientBundle.entry.push({resource: p});
-          });
-        } else {
-          patientBundle.entry.push({resource: pr});
+      // Record who is entering and submitting the responses
+      // How do we know the author here?
+      let questionnaireAuthor = process.env.VUE_APP_QUESTIONNAIRE_AUTHOR && process.env.VUE_APP_QUESTIONNAIRE_AUTHOR.toLowerCase();
+      if (questionnaireAuthor == 'practitioner') {
+        // Only add the `author` element if we can get the user id from the client
+        if (client.user && client.user.fhirUser) {
+          this.questionnaireResponse.author = {
+            reference: client.user.fhirUser
+          };
+        }
+      } else if (questionnaireAuthor == 'patient') {
+        this.questionnaireResponse.author = {
+          reference: this.questionnaireResponse.subject.reference
         }
       }
-    });
+    },
+    initializeSurveyObjEvents() {
+      // Add an event listener which updates questionnaireResponse based upon user responses
+      this.survey.onValueChanging.add(function(sender, options) {
+        // We don't want to modify anything if the survey has been submitted/completed.
+        if (sender.isCompleted == true) return;
+        
+        if (options.value != null) {
+          // Find the index of this item (may not exist)
+          // NOTE: THIS WON'T WORK WITH QUESTIONNAIRES THAT HAVE NESTED ITEMS
```

Comment:
If we ever want to consider nested surveys or branching, this is slightly troubling, but probably something we can work around